subprojects/GOPATH/src/github.com/microsoft/hivedscheduler/pkg/algorithm/cell.go

```diff
@@ -211,58 +248,78 @@ func (c *PhysicalCell) SetReserved(reserved bool) {
 	c.reserved = reserved
 }
 
+func (c *PhysicalCell) GetStatus() *api.PhysicalCellStatus {
+	return c.status
+}
+
 // VirtualCell defines a cell in a VC.
 type VirtualCell struct {
 	GenericCell
 	vc                   api.VirtualClusterName // name of its VC
 	rid                  api.ReservationId      // reservation ID
-	indexInChain         int32                  // index of the cell in the ChainCellList it belongs to (assigned in initialization)
 	preAssignedCell      *VirtualCell           // top level cell of this cell chain
 	physicalCell         *PhysicalCell          // points to the bound physical cell
 	preBoundPhysicalCell *PhysicalCell          // points to the temporarily bound physical cell (before the binding is confirmed)
+	status               *api.VirtualCellStatus // json representation of the status of this cell to expose to users
 }
```

Comment:
Just the "it only contains the cell status needed to be exposed to external, it should not be used as internal status management"? 
It is not related to json, it is just golang obj, and can be converted to yaml in future.